Add folderfilter config option

leophys opened the pull request:
I need this because one of the accounts I have misbehaves. Offlineimap tries to create some default IMAP folders that should yet exist...

This PR introduces a mechanism to filter out selected folder names, adding a `folderfilter` key to the account configuration. In every configuration spec chosen (json, ini, toml) it should be a list.

blallo committed:
Add folderfilter config option

boyska commented:
I think the right name for the option you are proposing is "excludefolders". That option should default to `["sieve"]` and be configurable if the user wants (even removing sieve from the list, if that's what the user wants).

Anothe problem I see is that I think your code works just because JSON syntax and python syntax overlap. But that's just calling for troubles when the first problem arises with quotes, spaces, escapes, whatever. Can you please improve it? I think it might be handy, in rewriting this, that you can use `mailbundle.json`. I'm not sure that will make things easier, but I think so.

blallo committed:
Improve rendering logic, rename and introduce custom func

leophys commented:
As we said off channel, we don't really know how jinja renders objects. We speculated it takes the `str` of them.

Anyways, I refactored the rendering logic and renamed the stuff following your suggestions.

I also added an `excludefolders_func` new option to allow one to specify a custom `folderfilter` func for `offlineimap`.
